src/json/formatter.cc

```diff
@@ -45,16 +46,70 @@ sstring formatter::end(state s) {
     }
 }
 
+static inline bool is_control_char(char c) {
+    return c >= 0 && c <= 0x1F;
+}
+
+static bool needs_escaping(const string_view& str) {
+    return ranges::any_of(str, [] (char c) {
```

Comment:
I'm not an expert but aren't ranges c+20? Some time ago Seastar was being kept at C++17. Has that changed @avikivity?

Reply:
This is true. Seastar is still supposed to support C++17...
I think sometimes we forget that loops exists ;-)

Reply:
Done